### PR TITLE
Added Trademarks check with acronym allowed

### LINKED ISSUE #464: Allow `wp-` prefixes on plugins
Per https://meta.trac.wordpress.org/ticket/5868 Using `WP ...` as the plugin name is allowed, but the trademark checks on WordPress.org block it preemptively.

Prefixes such as `"WP .."` and `"WC ..."` should not show as prohibited, but rather, note that they cannot be changed later to the trademarked term.

Plugin check may want to simply remove the `wp-` trademark check, leaving that to w.org/plugins, or may want to add a lesser-level note that it can't be changed to the trademark.

### PR DESCRIPTION
**Pull request by davidperezgar**
Fixes #464

### CODE REVIEW

File: includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php [hunk 1]
```diff
@@ -146,6 +147,18 @@ class Trademarks_Check extends Abstract_File_Check {
 		'you-tube-',
 	);
 
+	/**
+	 * Lists of trademarks that are allowed but shows in acronym.
+	 *
+	 * @since n.e.x.t.
```
Comment (ernilambar) [inline includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php]:
```suggestion
	 * @since 1.2.0
```

**Review by ernilambar — APPROVED** [review]

**Comment by davidperezgar** [comment]
There were [changes](https://wordpressfoundation.org/trademark-policy/#:~:text=The%20abbreviation%20%E2%80%9CWP%E2%80%9D%20is%20not%20covered%20by%20the%20WordPress%20trademarks%2C%20but%20please%20don%E2%80%99t%20use%20it%20in%20a%20way%20that%20confuses%20people.) in WordPress Foundation Trademark's, and we have to cover in this PR.

File: includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php [hunk 2]
```diff
@@ -348,6 +361,17 @@ private function validate_name_has_no_trademarks( $plugin_name ) {
...
 				esc_html( trim( $check, '-' ) )
 			);
+		} elseif (
+			trim( $check, '-' ) === $check
+			&& in_array( $check, self::FOR_USE_ACRONYM, true )
+		) {
+			// Trademarks that are allowed to use with Acronym.
+			$message = sprintf(
+				/* translators: 1: plugin slug, 2: found trademarked term */
+				__( 'The plugin name includes a restricted term. Your plugin name - "%1$s" - contains the restricted term "%2$s" which can be used within your plugin name, unless your plugin name will not change to full name. For example: You can use WP but not WordPress.', 'plugin-check' ),
```
Comment (swissspidy) [inline includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php]:
```suggestion
				__( 'The plugin name includes a restricted term. Your plugin name - "%1$s" - contains the restricted term "%2$s" which can be used , as long as you don\'t change it to the full name. For example: You can use WP but not WordPress.', 'plugin-check' ),
```

File: includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php [hunk 3]
```diff
@@ -348,6 +361,17 @@ private function validate_name_has_no_trademarks( $plugin_name ) {
 				esc_html( $plugin_name ),
 				esc_html( trim( $check, '-' ) )
 			);
+		} elseif (
+			trim( $check, '-' ) === $check
+			&& in_array( $check, self::FOR_USE_ACRONYM, true )
+		) {
+			// Trademarks that are allowed to use with Acronym.
```
Comment (swissspidy) [inline includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php]:
```suggestion
			// Trademarks that are allowed to use as an acronym.
```

File: includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php [hunk 4]
```diff
@@ -395,6 +419,17 @@ private function validate_slug_has_no_trademarks( $plugin_slug ) {
...
+			$message = sprintf(
+				/* translators: 1: plugin slug, 2: found trademarked term */
+				__( 'The plugin slug includes a restricted term. Your plugin slug - "%1$s" - contains the restricted term "%2$s" which can be used within in your plugin slug, unless your plugin name will not change to the full name. For example: You can use WP but not WordPress.', 'plugin-check' ),
```
Comment (swissspidy) [inline includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php]:
```suggestion
				__( 'The plugin slug includes a restricted term. Your plugin slug - "%1$s" - contains the restricted term "%2$s" which can be used within the plugin slug, as long as you don\'t use the full name in the plugin name. For example: You can use WP but not WordPress.', 'plugin-check' ),
```

File: tests/phpunit/tests/Checker/Checks/Trademarks_Check_Tests.php [hunk 5]
```diff
@@ -75,6 +75,12 @@ public function data_trademarks_check() {
 				'load.php',
 				'The plugin name includes a restricted term. Your chosen plugin name - "WooCommerce Example String" - contains the restricted term "woocommerce" which cannot be used within in your plugin name, unless your plugin name ends with "for woocommerce". The term must still not appear anywhere else in your name.',
 			),
+			'Plugin headers - WP Example String'          => array(
+				Trademarks_Check::TYPE_NAME,
+				'test-trademarks-plugin-header-acronym/load.php',
+				'load.php',
+				'The plugin name includes a restricted term. Your plugin name - "WP Example String" - contains the restricted term "wp" which can be used within your plugin name, unless your plugin name will not change to full name. For example: You can use WP but not WordPress.',
```
Comment (swissspidy) [inline tests/phpunit/tests/Checker/Checks/Trademarks_Check_Tests.php]:
```suggestion
				'The plugin name includes a restricted term. Your plugin name - "%1$s" - contains the restricted term "%2$s" which can be used , as long as you don\'t change it to the full name. For example: You can use WP but not WordPress.',
```

File: includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php [hunk 6]
```diff
@@ -146,6 +147,18 @@ class Trademarks_Check extends Abstract_File_Check {
...
 	);
 
+	/**
+	 * Lists of trademarks that are allowed but shows in acronym.
+	 *
+	 * @since 1.2.0
+	 *
+	 * @var string[]
+	 */
+	const FOR_USE_ACRONYM = array(
```
Comment (swissspidy) [inline includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php]: Maybe rename to `ALLOWED_ACRONYMS`? Makes it more clear.

File: includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php [hunk 7]
```diff
@@ -146,6 +147,18 @@ class Trademarks_Check extends Abstract_File_Check {
 		'you-tube-',
 	);
 
+	/**
+	 * Lists of trademarks that are allowed but shows in acronym.
```
Comment (swissspidy) [inline includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php]:
```suggestion
	 * Lists of allowed acronyms of trademarks.
```

File: includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php (hunk 6 above)
Comment (davidperezgar) [reply includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php]: ok

**Comment by davidperezgar** [comment]
Please don't merge yet as we need to clarify to the Foundation the trademark changes.

**Comment by davidperezgar** [comment]
Result of conversations:
> The current status-quo of not allowing wp- in slugs is fine, no need to change that, but if they use it in their plugin name that’s fine.

**Comment by davidperezgar** [comment]
After our Plugin Review Meeting, we were agreed that this PR can be published. Please review and accept it!

File: includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php [hunk 8]
```diff
@@ -146,6 +147,18 @@ class Trademarks_Check extends Abstract_File_Check {
 		'you-tube-',
 	);
 
+	/**
+	 * Lists of allowed acronyms of trademarks.
+	 *
+	 * @since 1.2.0
```
Comment (ernilambar) [inline includes/Checker/Checks/Plugin_Repo/Trademarks_Check.php]:
```suggestion
	 * @since 1.3.0
```

**Review by frantorres — APPROVED** [review]

**Pull request merged**